src/components/Api.js

```diff
@@ -0,0 +1,99 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._checkPromise);
+  }
+
+  _checkPromise(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return this._request(`${this._baseURL}/cards`, { headers: this._headers });
+  }
+
+  getUserInfo() {
+    return this._request(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    }).then((userData) => {
+      return userData;
+    });
+  }
+
+  updateUserInfo(name, about) {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        about,
+      }),
+    }).then(this._checkPromise);
+  }
+
+  //this doesn't work...have not been able to debug yet//
```

Comment:
> Or am I looking in the wrong place to fix this error

Please, delete `.then(this._checkPromise);` from each method to be able to use `this._request` (it contains inside it)


```js

  updateUserInfo(name, about) {
    return this._request(`${this._baseURL}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name,
        about,
      }),
    });
  }
```